test/specs/commonTests.js

```diff
@@ -0,0 +1,361 @@
+      .should.have.descendants(props)
+  })
+
+  // ----------------------------------------
+  // Defines _meta
+  // ----------------------------------------
+  describe('_meta', () => {
+    it('is a static object prop', () => {
+      expect(_meta).to.be.an('object')
+    })
+
+    describe('library', () => {
+      it('is defined', () => {
+        expect(_meta).to.have.any.keys('library')
+      })
+      it('is a META.library', () => {
+        expect(_.values(META.library)).to.include(_meta.library)
+      })
+    })
+    describe('name', () => {
+      it('is defined', () => {
+        expect(_meta).to.have.any.keys('name')
+      })
+      it('matches the filename', () => {
+        expect(_meta.name).to.equal(filenameWithoutExt)
+      })
+    })
+    if ('parent' in _meta) {
+      describe('parent', () => {
+        it('matches some component name', () => {
+          expect(_.map(stardust, c => c.prototype.constructor.name)).to.contain(_meta.parent)
+        })
+      })
+    }
+    describe('type', () => {
+      it('is defined', () => {
+        expect(_meta).to.have.any.keys('type')
+      })
+      it('is a META.type', () => {
+        expect(_.values(META.type)).to.include(_meta.type)
+      })
+    })
+  })
+
+  // ----------------------------------------
+  // Handles className
+  // ----------------------------------------
+  describe('className', () => {
+    it(`has some child with the "${sdClass}" class`, () => {
+      render(<Component {...requiredProps} />)
+        .should.have.className(sdClass)
+    })
+
+    it(`has the component className "${componentClassName}"`, () => {
+      render(<Component {...requiredProps} />)
+        .should.have.className(componentClassName)
+    })
+
+    it("applies user's className to root component", () => {
+      const classes = faker.hacker.phrase()
+      shallow(<Component {...requiredProps} className={classes} />)
+        .hasClass(classes)
+    })
+  })
+}
+
+/**
+ * Assert a component adds the Semantic UI "ui" className.
+ * @param {React.Component} Component The Component.
+ * @param {object} [props={}] A props object.
+ */
+export const hasUIClassName = (Component, props = {}) => {
+  it('has the "ui" className', () => {
+    shallow(createElement(Component, props))
+      .should.have.className('ui')
+  })
+}
+
+/**
+ * Assert a component renders children somewhere in the tree.
+ * @param {React.Component} Component A component that should render children.
+ * @param {object} [props={}] A props object.
+ */
+export const rendersChildren = (Component, props = {}) => {
+  describe('children', () => {
+    it('renders children', () => {
+      const child = faker.hacker.phrase()
+      shallow(createElement(Component, props, child))
+        .should.include.text(child)
+    })
+  })
+}
+
+// ----------------------------------------
+// className from prop
+// ----------------------------------------
+const _definesPropOptions = (Component, propName) => {
```

Comment:
There are a four private tests here that are re-used in the exported tests.   This allows us to mix and match these specific private tests to create our common className to prop tests below.